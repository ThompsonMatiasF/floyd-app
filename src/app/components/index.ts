export { default as ContainerCity } from '../containers/ContainerCity'
export { default as ContainerCurrentCity } from '../containers/ContainerCurrentCity'
export { default as City } from './City'
export { default as Weather } from './Weather'
export { default as WeatherIcon } from './WeatherIcon'
export { default as Forecast } from './Forecast'
export { default as ForecastDay } from './ForecastDay'
export { default as Tabs } from './Tabs'
export { default as Tab } from './Tab'
