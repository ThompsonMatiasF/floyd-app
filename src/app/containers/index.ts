export { default as ContainerCity } from './ContainerCity'
export { default as ContainerCurrentCity } from './ContainerCurrentCity'
