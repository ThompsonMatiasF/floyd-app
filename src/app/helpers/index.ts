export { default as constants } from './constants'
export { default as cityWeatherBuilder } from './weatherBuilder'
export { weatherClient } from './weatherClient'
