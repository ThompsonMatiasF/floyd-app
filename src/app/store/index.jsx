import React, { useReducer } from 'react'
import rootReducer from './reducers'

export const AppContext = React.createContext()

const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(...rootReducer);
  return <AppContext.Provider value={{state, dispatch}}>{children}</AppContext.Provider>
}

export default StoreProvider
