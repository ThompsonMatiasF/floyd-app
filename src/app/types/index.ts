export interface Temperature {
  main: number
  min: number
  max: number
}

export interface Weather {
  date: number
  description: string
  icon: string
  temp: Temperature
}

export interface Coordinates {
  lat: number
  lon: number
}

export interface Country {
  id: string
  name: string
}

export interface Location {
  id: number
  country: Country
  city: string
  coord: Coordinates
}

export interface CityWeather {
  location: Location
  weather: Weather
  forecast: Weather[]
}

export interface RequestState {
  LOADING: string
  SUCCESS: string
  FAILED: string
}

export interface WeatherActions {
  CITY: RequestState
  CURRENT_CITY: RequestState
  ACTIVE_TAB: string
}
