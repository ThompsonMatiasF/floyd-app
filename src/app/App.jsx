import React from 'react'
import '@axios-mock-db'
import StoreProvider from 'app/store'
import { Layout } from 'app/layout'
import moment from 'moment'
import 'moment/locale/es'

moment.locale('es')

const App = () => (
  <StoreProvider>
    <Layout />
  </StoreProvider>
);

export default App
