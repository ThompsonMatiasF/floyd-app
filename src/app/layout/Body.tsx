import React from 'react'
import { Tabs } from 'app/components'
import { ContainerCurrentCity, ContainerCity } from 'app/containers'

const Body = () => (
  <div className="flex flex-1 flex-wrap flex-col sm:flex-row md:flex-col justify-around bg-primary my-4 px-8 md:px-0">
    <ContainerCurrentCity />
    <span className="border-gray-200 mt-4 border-t-2 pt-8 mb-2 w-full" />
    <ContainerCity >
      <Tabs />
    </ContainerCity>
  </div>
)

export default Body
