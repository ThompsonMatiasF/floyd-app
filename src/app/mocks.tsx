import React, {FC} from 'react'
import { CityWeather, Location, Weather } from 'app/types'
import { initialState } from './store/reducers/weather/weather.reducer'
import { AppContext } from 'app/store';

export const weatherMock: Weather = {
  date: 1610553600,
  description: 'cielo claro',
  icon: '01d',
  temp:  {
    main: 26,
    min: 25,
    max: 27,
  },
}

export const locationMock: Location = {
  id: 3435010,
  country: {
    id: 'AR',
    name: 'Argentina',
  },
  city: 'Don Torcuato',
  coord: {
    lat: -34.4924,
    lon: -58.6162,
  },
}

export const cityWeatherMock: CityWeather = {
  location: locationMock,
  weather: weatherMock,
  forecast: Array(5).fill(weatherMock),
}

const state = {
  weather: {
    ...initialState,
    city: {
      ...initialState.city,
      data: cityWeatherMock,
    },
    currentCity: {
      ...initialState.currentCity,
      data: cityWeatherMock,
    },
  },
}

export const MockStoreProvider: FC = ({ children }) => (
  <AppContext.Provider value={{state, dispatch: jest.fn()}}>{ children }</AppContext.Provider>
)
