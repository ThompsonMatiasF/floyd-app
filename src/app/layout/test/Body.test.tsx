import { render } from '@testing-library/react'
import Body from '../Body'
import { MockStoreProvider } from 'app/mocks'

describe('Body.tsx', () => {
  it('should match snapshot', () => {
    const container = render(<MockStoreProvider><Body /></MockStoreProvider>).container

    expect(container).toMatchSnapshot()
  })
})
