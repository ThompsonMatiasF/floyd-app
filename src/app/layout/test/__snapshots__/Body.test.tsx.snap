// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body.tsx should match snapshot 1`] = `
<div>
  <div
    class="flex flex-1 flex-wrap flex-col sm:flex-row md:flex-col justify-around bg-primary my-4 px-8 md:px-0"
  >
    <div>
      <div
        class="flex flex-col md:flex-row justify-between items-stretch sm:items-center md:items-stretch rounded-b-2xl w-auto sm:w-1/2 md:w-auto"
      >
        <div
          class="flex flex-col items-center sm:items-stretch font-lg p-4 text-2xl flex-1 text-center md:text-left"
        >
          <span
            class="text-5xl font-bold"
          >
            Don Torcuato
            <span
              class="text-3xl font-light"
            >
              , 
              Argentina
            </span>
          </span>
          <span
            class="italic font-light flex-1 pt-2 font-bold capitalize"
          >
            jueves 14
          </span>
          <span
            class="font-bold uppercase"
          >
            cielo claro
          </span>
        </div>
        <div
          class="flex flex-row justify-center items-center bg-blue-500 pl-4 py-2 rounded-b-2xl"
        >
          <div
            class="flex flex-col justify-center font-bold text-center"
          >
            <span
              class="text-6xl md:text-9xl"
            >
              26
              º
            </span>
            <span
              class="text-xl"
            >
              25
              º / 
              27
              º
            </span>
          </div>
          <img
            alt="cielo claro"
            src="http://openweathermap.org/img/wn/01d@4x.png"
          />
        </div>
      </div>
      <div
        class="flex flex-row flex-wrap justify-between align-items-center w-auto sm:w-1/2 md:w-auto pl-0 sm:pl-4 md:pl-0"
      >
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="border-gray-200 mt-4 border-t-2 pt-8 mb-2 w-full"
    />
    <div>
      <div
        class="flex flex-row text-center justify-content-center w-full pb-2"
      >
        <button
          class="
        w-1/5 ml-2 mr-2 first:ml-0 last:mr-0 p-2 rounded-md 
        text-white font-bold
      "
          data-testid="tab"
          disabled=""
        >
          <span
            class="hidden sm:block rounded-b-"
          >
            Madrid
          </span>
          <span
            class="sm:hidden block"
          >
            ES
          </span>
        </button>
        <button
          class="
        w-1/5 ml-2 mr-2 first:ml-0 last:mr-0 p-2 rounded-md 
        text-gray-900 bg-white
      "
          data-testid="tab"
        >
          <span
            class="hidden sm:block rounded-b-"
          >
            New York
          </span>
          <span
            class="sm:hidden block"
          >
            US
          </span>
        </button>
        <button
          class="
        w-1/5 ml-2 mr-2 first:ml-0 last:mr-0 p-2 rounded-md 
        text-gray-900 bg-white
      "
          data-testid="tab"
        >
          <span
            class="hidden sm:block rounded-b-"
          >
            Paris
          </span>
          <span
            class="sm:hidden block"
          >
            FR
          </span>
        </button>
        <button
          class="
        w-1/5 ml-2 mr-2 first:ml-0 last:mr-0 p-2 rounded-md 
        text-gray-900 bg-white
      "
          data-testid="tab"
        >
          <span
            class="hidden sm:block rounded-b-"
          >
            Roma
          </span>
          <span
            class="sm:hidden block"
          >
            IT
          </span>
        </button>
        <button
          class="
        w-1/5 ml-2 mr-2 first:ml-0 last:mr-0 p-2 rounded-md 
        text-gray-900 bg-white
      "
          data-testid="tab"
        >
          <span
            class="hidden sm:block rounded-b-"
          >
            Tokyo
          </span>
          <span
            class="sm:hidden block"
          >
            JP
          </span>
        </button>
      </div>
      <div
        class="flex flex-col md:flex-row justify-between items-stretch sm:items-center md:items-stretch rounded-b-2xl w-auto sm:w-1/2 md:w-auto"
      >
        <div
          class="flex flex-col items-center sm:items-stretch font-lg p-4 text-2xl flex-1 text-center md:text-left"
        >
          <span
            class="text-5xl font-bold"
          >
            Don Torcuato
            <span
              class="text-3xl font-light"
            >
              , 
              Argentina
            </span>
          </span>
          <span
            class="italic font-light flex-1 pt-2 font-bold capitalize"
          >
            jueves 14
          </span>
          <span
            class="font-bold uppercase"
          >
            cielo claro
          </span>
        </div>
        <div
          class="flex flex-row justify-center items-center bg-blue-500 pl-4 py-2 rounded-b-2xl"
        >
          <div
            class="flex flex-col justify-center font-bold text-center"
          >
            <span
              class="text-6xl md:text-9xl"
            >
              26
              º
            </span>
            <span
              class="text-xl"
            >
              25
              º / 
              27
              º
            </span>
          </div>
          <img
            alt="cielo claro"
            src="http://openweathermap.org/img/wn/01d@4x.png"
          />
        </div>
      </div>
      <div
        class="flex flex-row flex-wrap justify-between align-items-center w-auto sm:w-1/2 md:w-auto pl-0 sm:pl-4 md:pl-0"
      >
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
        <div
          class="w-full md:w-auto flex flex-row items-center my-4"
          data-testid="forecast-day"
          style="border-radius: 50px;"
        >
          <div
            class="bg-blue-500"
            style="border-radius: 50px;"
          >
            <img
              alt="cielo claro"
              src="http://openweathermap.org/img/wn/01d.png"
            />
          </div>
          <div
            class="text-center flex-1 pl-4 pr-6 text-xl md:text-base"
          >
            <div
              class="inline-block md:block capitalize"
            >
              Wednesday 13
            </div>
            <span
              class="inline md:hidden"
            >
               - 
            </span>
            <div
              class="inline-block md:block"
            >
              <span
                class="font-bold"
              >
                25
                º
              </span>
               / 
              <span
                class="font-bold"
              >
                27
                º
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
